app/src/main/java/de/danoeh/antennapod/activity/MainActivity.java

```diff
@@ -439,6 +441,17 @@ protected void onResume() {
             finish();
             startActivity(new Intent(this, MainActivity.class));
         }
+
+        handleHiddenScreen();
+    }
+
+    public void handleHiddenScreen() {
+        String lastNavFragment = NavDrawerFragment.getLastNavFragment(this);
+        List<String> hiddenDrawerItems = UserPreferences.getHiddenDrawerItems();
+        if (hiddenDrawerItems.contains(lastNavFragment)) {
+            String defaultPage = UserPreferences.getDefaultPage();
+            new MainActivityStarter(this).withFragmentLoaded(defaultPage).start();
+        }
```

Comment:
My idea was to call something like this code from the DrawerPreferencesDialog. That way, we do not need a cyclic dependency from the dialog to the activity.

Reply:
Moving the code to the DrawerPreferencesDialog will result in the user being redirected to the MainActivity once the confirm button is clicked. This might not be ideal in situations where the user what to make more changes on the settings page. Will this behaviour be acceptable?

Reply:
Hmm, right. I will need to think about the code structure here a bit.